### DISCUSSION TITLE
Make row collapsed option a url parameter

### DISCUSSION THREAD

**Discussion by kevisong** [grafana/grafana#38859] [Dashboards, Panels & Visualizations]
## Proposal

Make row collapsed option a url parameter.

## Current Behavior

Row collapsed or not can only be saved in the dashboard JSON.

## Expected Behavior

Row collapsed or not can be modified via url parameter just like variable.

Below is an example with url parameter `panel-HTTP-collapsed` set to true.

```
http://grafana-app.com/d/v6qxizV7k/Overview?orgId=1&refresh=5m&panel-HTTP-collapsed=true
```

And the row named HTTP in the dashboard will not be collapsed when opened via above URL.

![image](https://user-images.githubusercontent.com/39701004/132086172-8ed302e2-ebe0-403c-9d25-1f9701f2d343.png)

When using below url:

```
http://grafana-app.com/d/v6qxizV7k/Overview?orgId=1&refresh=5m&panel-HTTP-collapsed=false
```

The row will be collapsed when the dashboard is opened.

## Usage

In a situation when having a dashboard packed with a number of monitoring entities separated by different row. In addition, there is a backend server that automatically generates the URL based on the user's role.

What what can be achieved is that the backend server can automatically generate the URL with collapsed rows that the user's role is not interested in by dynamically control the collapsed option in the url parameter.

**Comment by jeff-zone** [comment]
Several years later, this is still relevant.  Any workaround to achieve this behavior using URL variables?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82973. Feel free to continue the discussion around this there. Thank you!
